In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings(action='ignore')

In [2]:
data = pd.read_csv('data/data.csv', encoding= 'unicode_escape')

before_corona = data[data['end_date'] < '2020-03-17']
after_corona = data[data['end_date'] > '2020-03-17']

In [3]:
#필요없는 column 제거
before_corona.drop(['id', 'uri', 'track_href', 'analysis_url', 'type'], axis=1, inplace=True)
after_corona.drop(['id', 'uri', 'track_href', 'analysis_url', 'type'], axis=1, inplace=True)

In [4]:
before_corona.head()

Position       Track Name       Artist   Streams  region  \
0         1     Dance Monkey  Tones And I  46784119  global   
1         2          The Box  Roddy Ricch  46163456  global   
2         3  Blinding Lights   The Weeknd  39499306  global   
3         4             Tusa      KAROL G  34306204  global   
4         5  Don't Start Now     Dua Lipa  31619914  global   

               spotify_id  danceability  energy  key  loudness  ...  \
0  1rgnBhdG2JDFTbYkYRZAku         0.825   0.593    6    -6.401  ...   
1  0nbXyq5TXYPCO7pr3N8S4I         0.896   0.586   10    -6.687  ...   
2  0sf12qNH5qcw8qpgymFOqD         0.513   0.796    1    -4.075  ...   
3  7k4t7uLgtOxPwTpFmtJNTY         0.803   0.715    2    -3.280  ...   
4  6WrI0LAC5M1Rw2MnX2ZvEg         0.794   0.793   11    -4.521  ...   

   speechiness  acousticness  instrumentalness  liveness  valence    tempo  \
0       0.0988       0.68800          0.000161    0.1700    0.540   98.078   
1       0.0559       0.10400          0.000000    0.7900    0.642  116.971   
2       0.0629       0.00147          0.000209    0.0938    0.345  171.017   
3       0.2980       0.29500          0.000134    0.0574    0.574  101.085   
4       0.0842       0.01250          0.000000    0.0952    0.677  123.941   

   duration_ms  time_signature  start_date    end_date  
0       209755               4  2020-01-31  2020-02-07  
1       196653               4  2020-01-31  2020-02-07  
2       201573               4  2020-01-31  2020-02-07  
3       200960               4  2020-01-31  2020-02-07  
4       183290               4  2020-01-31  2020-02-07  

[5 rows x 21 columns]

In [5]:
print('before_corona shape :{}'.format(before_corona.shape))
print('after_corona shape :{}'.format(after_corona.shape))

before_corona shape :(1200, 21)
after_corona shape :(2400, 21)


* 코로나 이전에 1200개의 데이터, 코로나 이후에 2400개의 데이터가 존재
* 중복값을 제거해 unique value만 남김

In [6]:
before_corona.drop_duplicates(subset='spotify_id', inplace=True)
after_corona.drop_duplicates(subset='spotify_id', inplace=True)

In [7]:
print('before_corona shape :{}'.format(before_corona.shape))
print('after_corona shape :{}'.format(after_corona.shape))

before_corona shape :(345, 21)
after_corona shape :(423, 21)


코로나 이전에 345개의 음악, 코로나 이후에 423개의 음악이 순위권에 오름

In [8]:
before_corona.columns

Index(['Position', 'Track Name', 'Artist', 'Streams', 'region', 'spotify_id',
       'danceability', 'energy', 'key', 'loudness', 'mode', 'speechiness',
       'acousticness', 'instrumentalness', 'liveness', 'valence', 'tempo',
       'duration_ms', 'time_signature', 'start_date', 'end_date'],
      dtype='object')

In [9]:
print(before_corona['danceability'].mean())
print(after_corona['danceability'].mean())

0.7090811594202897
0.7130047281323876


In [ ]:
print(before_corona['energy'].mean())
print(after_corona['energy'].mean())

In [ ]:
print(before_corona['danceability'].mean())
print(after_corona['danceability'].mean())